ext/bg/js/search.js

```diff
@@ -166,13 +162,12 @@ class DisplaySearch extends Display {
     }
 
     onPopState() {
-        const query = DisplaySearch.getSearchQueryFromLocation(window.location.href) || '';
-        if (this.query !== null) {
-            if (this.isWanakanaEnabled()) {
-                this.setQuery(window.wanakana.toKana(query));
-            } else {
-                this.setQuery(query);
-            }
+        const {query='', mode=''} = DisplaySearch.parseQueryStringFromLocation(window.location.href);
+        document.documentElement.dataset.searchMode = mode;
+        if (this.isWanakanaEnabled()) {
```

Comment:
Could be simplified
```js
this.setQuery(this.isWanakanaEnabled() ? window.wanakana.toKana(query) : query);
```

Reply:
This is also done on line 206, and could also be simplified on line 83. Maybe update `setQuery` to do this automatically? It doesn't look like there's a current need for this, but a `setQueryRaw` or something similar could be added if necessary to have one which doesn't modify input; or a second `raw` (or something) parameter to `setQuery`.

Reply:
I updated it to line 206 and also to the checkbox event listener because the value is checked directly from the checkbox